test/unit/Encoding/MicrosecondBasedDateConversionTest.php

```diff
@@ -35,17 +34,16 @@ public function dateClaimsHaveMicrosecondsOrSeconds(): void
         $formatter = new MicrosecondBasedDateConversion();
         $formatted = $formatter->formatClaims($claims);
 
-        self::assertSame(1487285080, $formatted[RegisteredClaims::ISSUED_AT]);
-        self::assertSame('1487285080.000123', $formatted[RegisteredClaims::NOT_BEFORE]);
-        self::assertSame('1487285080.123456', $formatted[RegisteredClaims::EXPIRATION_TIME]);
+        self::assertSame(1487285080.0, $formatted[RegisteredClaims::ISSUED_AT]);
```

Comment:
This is a BC break for those who are relying right now to strict check to integers, isn't it?

Reply:
The interface uses DateTimeImmutable. The BC-break is with the encoded token (which, as reported many times, shouldn't never be a string).

Reply:
We can keep the int, not an issue